pallets/api/src/extension.rs

```diff
@@ -0,0 +1,54 @@
+use core::marker::PhantomData;
```

Comment:
This modules aims to store anything that is both specifc to pop-api and reusable for chain extensions implementations across runtimes.

e.g.
[contract envinroment: contract <> `pop-api` ] <----> [runtime: `pop-chain-extension` <> `pallet-api` <> `fungibles`]